src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -260,6 +261,27 @@ public String getName() {
     return this.name;
   }
 
+  /**
+   * Return the latest committed/persisted offset token for all channels
+   *
+   * @return map of channel to the latest persisted offset token
+   */
+  @Override
+  public Map<SnowflakeStreamingIngestChannel, String> getLatestCommittedOffsetTokens(
+      List<SnowflakeStreamingIngestChannel> channels) {
+    List<SnowflakeStreamingIngestChannelInternal<?>> internalChannels =
+        channels.stream()
+            .map(c -> (SnowflakeStreamingIngestChannelInternal<?>) c)
+            .collect(Collectors.toList());
+    List<ChannelsStatusResponse.ChannelStatusResponseDTO> channelsStatus =
+        getChannelsStatus(internalChannels).getChannels();
+    Map<SnowflakeStreamingIngestChannel, String> result = new HashMap<>();
+    for (int idx = 0; idx < channels.size(); idx++) {
+      result.put(channels.get(idx), channelsStatus.get(idx).getPersistedOffsetToken());
+    }
+    return result;
```

Comment:
In your use case, a Map makes more sense then just a list of offset tokens?

Reply:
Mmmm, if we guarantee ordering then a list is fine, but a map is nicer because the results don't have to be Zipped back together. Ideally we could return a list of some channel identifier to the offset token, maybe channel qualified name?

Reply:
I'm fine with either option, but using the fully qualified channel name sounds better?